src/app/shared/guards/pending-changes-guard.ts

```diff
@@ -11,6 +12,9 @@ export interface PendingChangesComponent {
   isDirty(): boolean,
 }
 
+const dialogResponse = new BehaviorSubject<'pending' | 'accepted' | 'declined'>('pending');
```

Comment:
For 3 cases like that, I prefer you define a enum that using the strings. Strings are just good as they can be used in templates as well, but here you do not need this case.